public/omni/cluster-management/scale-your-cluster/karpenter.mdx

```diff
@@ -0,0 +1,450 @@
+
+```yaml
+# cluster-template.yaml
+
+kind: Cluster
+name: <cluster-name> # Replace with your cluster name
+kubernetes:
+  version: v1.34.1 # Replace this version with your preferred version of Kubernetes
+talos:
+  version: v1.11.3 # Replace this version with your preferred version of Kubernetes
+
+---
+kind: ControlPlane
+machineClass:
+  name: <name of your control plane machine class>
+  size: <number of control plane machines>
+
+---
+kind: Workers
+machineClass:
+  name: <name of your worker machine class>
+  size: Unlimited
+```
+
+This template creates a cluster and assigns machines to the appropriate Machine Classes.
+Replace the following placeholders:
+
+- `<cluster-name>`— the name you want to give your cluster
+- `<name of your control plane machine class>`— the Machine Class for your control plane nodes.
+- `<number of control plane machines>`— how many control plane nodes your cluster should have.
+- `<name of your worker plane machine class>`— the Machine Class for your worker nodes.
+
+2. Apply the `cluster-template.yaml` to create the cluster:
+
+```bash
+omnictl cluster template sync -f cluster-template.yaml
+```
+
+3. Once your cluster is up, download and export your cluster's `kubeconfig` to interact with the cluster:
+
+    a. Select your cluster on the Omni dashboard.
+
+    b. Click **Download kubeconfig**.
+
+    c. Export the `kubeconfig`:
+
+    ```
+    export KUBECONFIG=<path to your kubeconfig>
+    ```
+
+4. Verify that all machines are running::
+
+```bash
+kubectl get nodes
+```
+
+## Step 5: Define all variables
+
+Set the environment variables you will use throughout this guide:
+
+```bash
+export CLUSTER_NAME="<cluster-name>"                # Your cluster name
+export AWS_DEFAULT_REGION="<aws-region>"            # For example: eu-west-1
+export TALOS_AMI_ID="<worker-ami-id>"               # For example: ami-0123456789abcdef0
+export SUBNET_IDS="subnet-xxxx subnet-yyyy"         # Subnets where nodes will launch
+export SECURITY_GROUP_IDS="sg-xxxx sg-yyyy"         # Security groups for worker nodes
+
+export KARPENTER_NAMESPACE="kube-system"
+export KARPENTER_VERSION="1.8.1"
+
+export CLUSTER_ENDPOINT="https://<your-endpoint>.omni.siderolabs.io"  # Your Omni API endpoint
+```
+
+**Why this matters**: Defining these variables once makes the rest of the guide copy-and-paste friendly.
+
+## Step 6: Tag subnets and security groups for Karpenter discovery
+
+Karpenter can only launch nodes into subnets and security groups that you explicitly mark for discovery.  
+You do this by adding a `karpenter.sh/discovery` tag to each resource.
+
+Run the following commands to tag your subnets and security groups:
+
+```bash
+aws ec2 create-tags \
+  --resources $SUBNET_IDS \
+  --tags Key=karpenter.sh/discovery,Value=$CLUSTER_NAME
+
+aws ec2 create-tags \
+  --resources $SECURITY_GROUP_IDS \
+  --tags Key=karpenter.sh/discovery,Value=$CLUSTER_NAME
+```
+
+## Step 7: Fix IAM permissions on the worker role
```

Comment:
Would it make more sense to run Karpenter on the control plane nodes so they don't need dedicated workers that aren't created via karpenter?